xdsl/ir.py

```diff
@@ -144,9 +144,14 @@ def name(self) -> str | None:
         return self._name
 
     @name.setter
-    def name(self, name: str):
-        if self._name_regex.fullmatch(name):
+    def name(self, name: str | None):
+        if name is None or self._name_regex.fullmatch(name):
             self._name = name
+        else:
+            raise ValueError(
+                "Invalid SSA Value name format!",
+                r"Allowed values must match [0-9]+|([A-Za-z_$.-][\w$.-]*)",
```

Comment:
I am not sure how understandable this error message is, Do we want people to have to understand python regex? Maybe this and a textual description of the allowed names would be a good idea.